### PR TITLE
Fix 0 based 1 based

### PR DESCRIPTION
**Pull request by lldelisle**
This PR was build on top of #17 
It focuses on the bed which are 0-based half-open and the gff which are 1-based closed.
This confusion was causing a lot of ambiguous read assignment.

### CODE REVIEW

**Comment by lldelisle** [comment]
I think I still have a bug with 0-coverage. If it is included into 0-coverage region, it generates a UTR of size 1. I need to change that.

**Comment by haessar** [comment]
Sorry I've not looked at this yet, will conclude reviewing early next week

**Comment by haessar** [comment]
@lldelisle. should the 1-based adjustment not be dependent on whether strand is forward or reverse? I notice there is a blanket "start + 1" regardless, which leads to reverse UTRs effectively being 1 base shorter, but still potentially overlapping boundary with the transcript at the UTRs end base.

Also, I'm not sure what the test-case3 is intended for as it appears that current and expected peaks2utr output are the same (is the current representing peaks2utr with your changes in place?).

File: peaks2utr/collections.py [hunk 1]
```diff
@@ -99,10 +99,16 @@ def filter(self, **kwargs):
...
     """
     class Interval:
+        """
+        1-based included
+        init from 0-based half-opened
+        """
         def __init__(self, start, end):
-            self.start = int(start)
+            self.start = int(start) + 1
             self.end = int(end)
```
[the remark this answers is not in the document]
Comment (lldelisle) [reply peaks2utr/collections.py:112]: GFF are 1-based closed intervals so 1-2 means 2 bases the first one and the second one.
BED are 0-based half-opened intervals so 1-2 means 1 base, the second one.
So to convert the BED interval 1-2 to GFF I need to add 1 to start but not touch the end: 2-2

**Comment by lldelisle** [comment]
> Also, I'm not sure what the test-case3 is intended for as it appears that current and expected peaks2utr output are the same (is the current representing peaks2utr with your changes in place?).

The test3 output has been changed in commit https://github.com/haessar/peaks2utr/pull/19/commits/09dd2c4572df0ce05b01a2c28a5356fdfce50a46 after the changes being introduced.

**Comment by lldelisle** [comment]
> @lldelisle. should the 1-based adjustment not be dependent on whether strand is forward or reverse? I notice there is a blanket "start + 1" regardless, which leads to reverse UTRs effectively being 1 base shorter, but still potentially overlapping boundary with the transcript at the UTRs end base.

As explained above, there are 2 changes between BED and GFF. The 0-based vs 1-based but also the 'half-opened' vs 'closed'. That's why it is relevant to only change the start coordinate independently of the strand.

Tell me if it is unclear.

**Comment by lldelisle** [comment]
Sorry. It took me a while to merge your PR...

**Comment by haessar** [comment]
Will fix linting in separate commit

**Pull request merged**